docs/docsite/rst/developing_modules_general_aci.rst

```diff
@@ -0,0 +1,1851 @@
+For more information about Cisco ACI, look at the `Cisco ACI user guide <https://www.cisco.com/c/en/us/solutions/collateral/data-center-virtualization/application-centric-infrastructure/solution-overview-c22-741487.html>`_.
+
+What's covered in this section:
+
+.. contents::
+  :depth: 3
+  :local:
+
+.. _aci_dev_guide_intro:
+
+Introduction
+============
+The `cisco.aci collection <https://galaxy.ansible.com/cisco/aci>`_ already includes a large number of Cisco ACI modules; however, the ACI object model is huge, and covering all possible functionality would easily require more than 1,500 individual modules. Therefore, Cisco develops modules requested by people on a just-in-time basis.
+
+If you need a specific functionality, you have 3 options:
+
+- Open an issue using https://github.com/CiscoDevNet/ansible-aci/issues/new/choose so that Cisco developers can build, enhance, or fix the modules for you.
+- Learn the ACI object model and use the low-level APIC REST API using the `aci_rest <https://docs.ansible.com/ansible/latest/collections/cisco/aci/aci_rest_module.html>`_ module.
+- Contribute to Cisco's ansible-aci project by writing your own dedicated modules, proposing a fix or an enhancement, and being part of the Cisco ansible-aci community.
+
+.. _aci_dev_guide_git:
+
+In this guide, we're going to concentrate on the third option to show you how to build your own module, fix an issue, or improve an existing module and contribute it back to the ansible-aci project. The first step in the process is to retrieve the latest version of the ansible-aci collection code.
+By retrieving the latest version, you will be able to modify existing code.
+
+Fork, Clone and Branch
+======================
+The collection code is located in a git repository (https://github.com/CiscoDevNet/ansible-aci). You can directly clone this repository to retrieve the latest version of the code, but in order to later contribute your code back to the project, you will need to create a fork to be able to create a proper pull request.
+
+**Fork**
+  A fork is a copy of a repository that allows you to make changes to the repository without affecting the original project. 
+
+You can contribute your changes back to the original project by using Pull Requests from the forked repository.
+
+Let's create a fork of the repository.
+
+* Go to: https://github.com/CiscoDevNet/ansible-aci
+* Fork CiscoDevNet's **ansible-aci** repo by clicking the top right-hand corner fork button.
+
+.. note::
+
+  Refer to - `How to fork a repo <https://docs.github.com/en/github/getting-started-with-github/fork-a-repo>`_
+
+Now that we have forked our repository, let's clone the forked repository on our local machine.
+
+**Clone**  
+  Clone allows you to copy a repository to your local machine.
+
+* Clone the forked repo by going to the terminal and enter the following command:
+
+.. code-block:: Blocks
+
+  git clone https://github.com/<Forked Organization>/ansible-aci.git
+
+**Naming Convention**
```

Comment:
not sure I understand why this chapter is called naming convention, when it is more related to remote/local repositories